packages/next/src/app-dir/links/nextCache.ts

```diff
@@ -18,6 +18,11 @@ import { generateCacheTag } from '../shared';
 
 type NextCacheLinkOptions<TRouter extends AnyRouter> = {
   router: TRouter;
+  /**
+   * define which values from the context should be considered into the cache
+   * key
+   */
+  cacheContext?: (ctx: inferRouterContext<TRouter>) => any[];
```

Comment:
I suggest making this property `| undefined` instead of partial so that user is explicitly passing `undefined` if he doesn't want to manipulate the cache key.

Reply:
included in the example!